pkg/model/application_live_state_test.go

```diff
@@ -168,3 +168,78 @@ func TestApplicationLiveStateSnapshot_DetermineAppHealthStatus(t *testing.T) {
 		})
 	}
 }
+
+func TestApplicationLiveStateSnapshot_DetermineApplicationHealthStatus(t *testing.T) {
+	testcases := []struct {
+		name     string
+		snapshot *ApplicationLiveStateSnapshot
+		want     ApplicationLiveStateSnapshot_Status
+	}{
+		{
+			name: "healthy: all resources are healthy",
+			snapshot: &ApplicationLiveStateSnapshot{
+				ApplicationLiveState: &ApplicationLiveState{
+					Resources: []*ResourceState{
+						{HealthStatus: ResourceState_HEALTHY},
+						{HealthStatus: ResourceState_HEALTHY},
+					},
+				},
+			},
+			want: ApplicationLiveStateSnapshot_HEALTHY,
+		},
+		{
+			name: "healthy: no resources",
+			snapshot: &ApplicationLiveStateSnapshot{
+				ApplicationLiveState: &ApplicationLiveState{},
+			},
+			want: ApplicationLiveStateSnapshot_HEALTHY,
```

Comment:
What happen if ApplicationLiveState = nil, not a dummy pointer?

Reply:
@khanhtc1202 
The case is `ApplicationLiveStateSnapshot_UNKNOWN` as the last of the testcases, similar to the current implementation.

Reply:
Thanks for clarify, commented in https://github.com/pipe-cd/pipecd/pull/5454/files#r1898201564